openflights.js

```diff
@@ -1715,94 +1724,62 @@ function showTop10(str) {
-  } else {
-    var master = str.split("\n");
-    var routes = master[0];
-    var airports = master[1];
-    var airlines = master[2];
-    var planes = master[3];
-    bigtable = "<table style='width: 100%; border-collapse: collapse'><td style='vertical-align: top; padding-right: 10px'><img src='/img/close.gif' onclick='JavaScript:closePane();' width=17 height=17><form id='top10form'>";
-    table = "<br>" + gt.gettext("Show...") + "<br>";
-    table += createSelectFromArray('limit', toplimits, "updateTop10()", limit) + "<br>";
-    table += gt.gettext("Sort by...") + "<br>";
-    table += createSelectFromArray('mode', topmodes, "updateTop10()", mode) + "<br>";
-    bigtable += table + "</form></td>";
-
-    bigtable += "<td style='vertical-align: top; background-color: #ddd; padding: 0px 10px'>";
-    table = "<table><tr><th colspan=3'>" + gt.gettext("Routes") + "</th></tr>";
-    var rows = routes.split(":");
-    for (r = 0; r < rows.length; r++) {
-      var col = rows[r].split(",");
-      // s.name, s.apid, d.name, d.apid, count
-      table += "<tr><td><a href='#' onclick='JavaScript:selectAirport(" + col[1] + ");'>" + col[0] + "</a>&harr;" +
-	"<a href='#' onclick='JavaScript:selectAirport(" + col[3] + ");'>" + col[2] + "</a></td>" + 
-	"<td style='text-align: right; padding-left: 10px'>" + parseInt(col[4]) + "</td></tr>";
-    }
-    table += "</table>";
+  bigtable = "<table style='width: 100%; border-collapse: collapse'><td style='vertical-align: top; padding-right: 10px'><img src='/img/close.gif' onclick='JavaScript:closePane();' width=17 height=17><form id='top10form'>";
+  table = "<br>" + gt.gettext("Show...") + "<br>";
+  table += createSelectFromArray('limit', toplimits, "updateTop10()", limit) + "<br>";
+  table += gt.gettext("Sort by...") + "<br>";
+  table += createSelectFromArray('mode', topmodes, "updateTop10()", mode) + "<br>";
+  bigtable += table + "</form></td>";
 
-    bigtable += table + "</td><td style='vertical-align: top; padding: 0px 10px'>";
-    table = "<table><tr><th colspan=3'>" + gt.gettext("Airports") + "</th></tr>";
-    var rows = airports.split(":");
-    for (r = 0; r < rows.length; r++) {
-      var col = rows[r].split(",");
-      var noOfColumns = col.length;
-      // name, iata, count, apid
-      if(noOfColumns > 4) {
-        var tooManyColumns = noOfColumns - 4;
-        // join name back together
-        var airportName = col.slice(0, tooManyColumns + 1).join(',');
-        desc = airportName.substring(0,20) + " (" + col[noOfColumns - 3] + ")";
-      }
-      else {
-        desc = col[0].substring(0,20) + " (" + col[1] + ")";
-      }
-      table += "<tr><td><a href='#' onclick='JavaScript:selectAirport(" + col[noOfColumns-1] + ");'>" + desc + "</a></td><td style='text-align: right; padding-left: 10px'>" + parseInt(col[noOfColumns-2]) + "</td>";
-    }
-    table += "</table>";
+  bigtable += "<td style='vertical-align: top; background-color: #ddd; padding: 0px 10px'>";
```

Comment:
I kept the HTML generation the same but I actually have a separate branch converted to use `document.createElement`. Take a look at https://github.com/chrisrosset/openflights/compare/top10-json...chrisrosset:openflights:rewrite-showtop10?expand=1 if you're interested in what that looks like.